File: back-end/mercado-inteligente-spring/src/main/java/br/edu/usf/mercadointeligentespring/gateways/database/repository/UserRepository.java
package br.edu.usf.mercadointeligentespring.gateways.database.repository;

import br.edu.usf.mercadointeligentespring.gateways.dto.UserDTO;
import org.springframework.data.repository.CrudRepository;
import org.springframework.stereotype.Repository;

@Repository
public interface UserRepository extends CrudRepository<UserDTO, Long> {

}

File: back-end/mercado-inteligente/src/main/kotlin/br/edu/usf/gateways/database/repository/UserRepository.kt
package br.edu.usf.gateways.database.repository

import br.edu.usf.gateways.dto.UserDTO
import io.quarkus.hibernate.orm.panache.kotlin.PanacheRepository
import javax.enterprise.context.ApplicationScoped

@ApplicationScoped
class UserRepository : PanacheRepository<UserDTO> {
    fun create(user: UserDTO): UserDTO {
        persist(user)
        return user;
    }

    fun update(id: Long, user: UserDTO) {
        user.id = id
        update("where id = ?1", id, user)
    }
}